### PR TITLE
[Merged by Bors] - feat: port Topology.UniformSpace.Cauchy

### PR DESCRIPTION
**Pull request by mcdoll**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
Checked out, fixing.

File: Mathlib/Topology/UniformSpace/Cauchy.lean [hunk 1]
```diff
@@ -0,0 +1,772 @@
...
+  exacts[⟨m, hm, refl_mem_uniformity ha⟩, ⟨n, le_refl n, hn m hm n le_rfl⟩]
+#align cauchy_seq.totally_bounded_range CauchySeq.totallyBounded_range
+
+/-!
+### Sequentially complete space
+
+In this section we prove that a uniform space is complete provided that it is sequentially complete
+(i.e., any Cauchy sequence converges) and its uniformity filter admits a countable generating set.
+In particular, this applies to (e)metric spaces, see the files `topology/metric_space/emetric_space`
+and `topology/metric_space/basic`.
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Cauchy.lean:619]:
```suggestion
In particular, this applies to (e)metric spaces, see the files `Topology/MetricSpace/EmetricSpace`
and `Topology/MetricSpace/Basic`.
```

File: Mathlib/Topology/UniformSpace/Cauchy.lean [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+
+theorem setSeq_prod_subset {N m n} (hm : N ≤ m) (hn : N ≤ n) :
+    setSeq hf U_mem m ×ˢ setSeq hf U_mem n ⊆ U N := fun p hp => by
+  refine' (setSeqAux hf U_mem N).2.2 ⟨_, _⟩ <;> apply setSeq_sub_aux
+  exact setSeq_mono hf U_mem hm hp.1
+  exact setSeq_mono hf U_mem hn hp.2
+#align sequentially_complete.set_seq_prod_subset SequentiallyComplete.setSeq_prod_subset
+
+/-- A sequence of points such that `seq n ∈ set_seq n`. Here `set_seq` is an antitone
+sequence of sets `set_seq n ∈ f` with diameters controlled by a given sequence
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Cauchy.lean:668]:
```suggestion
/-- A sequence of points such that `seq n ∈ setSeq n`. Here `setSeq` is an antitone
sequence of sets `setSeq n ∈ f` with diameters controlled by a given sequence
```

File: Mathlib/Topology/UniformSpace/Cauchy.lean [hunk 3]
```diff
@@ -0,0 +1,772 @@
...
+  choose_spec <| hf.1.nonempty_of_mem (setSeq_mem hf U_mem n)
+#align sequentially_complete.seq_mem SequentiallyComplete.seq_mem
+
+theorem seq_pair_mem ⦃N m n : ℕ⦄ (hm : N ≤ m) (hn : N ≤ n) :
+    (seq hf U_mem m, seq hf U_mem n) ∈ U N :=
+  setSeq_prod_subset hf U_mem hm hn ⟨seq_mem hf U_mem m, seq_mem hf U_mem n⟩
+#align sequentially_complete.seq_pair_mem SequentiallyComplete.seq_pair_mem
+
+-- include U_le -- Porting note: includes no longer needed
+
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Cauchy.lean:684]:
```suggestion
```
I don't think we'll return to this, so we don't need a porting note.

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Topology/UniformSpace/Cauchy.lean (hunk 3 above)
Comment (urkud) [inline Mathlib/Topology/UniformSpace/Cauchy.lean:684]:
```suggestion
```

**Comment by urkud** [comment]
bors merge

File: Mathlib/Topology/UniformSpace/Cauchy.lean [hunk 4]
```diff
@@ -0,0 +1,770 @@
...
+noncomputable section
+
+/-- An auxiliary sequence of sets approximating a Cauchy filter. -/
+def setSeqAux (n : ℕ) : { s : Set α // s ∈ f ∧ s ×ˢ s ⊆ U n } :=
+  -- Porting note: changed `∃ _ : s ∈ f, ..` to `s ∈ f ∧ ..`
+  indefiniteDescription _ <| (cauchy_iff.1 hf).2 (U n) (U_mem n)
+#align sequentially_complete.set_seq_aux SequentiallyComplete.setSeqAux
+
+/-- Given a Cauchy filter `f` and a sequence `U` of entourages, `set_seq` provides
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Cauchy.lean:642]: Oops
```suggestion
/-- Given a Cauchy filter `f` and a sequence `U` of entourages, `setSeq` provides
```

**Pull request closed without merging**